src/main/java/twitterapp/src/services/TwitterAppService.java

```diff
@@ -145,7 +155,32 @@ public TwitterAppService(Twitter twitter){
 
     }
 
+    public Optional<List<TwitterPost>> getUserTimeline() throws TwitterAppException{
+        try {
+            if(cacheUserTimeline.get(TIMELINE_KEY).isPresent() == false){
+                Paging page = new Paging(1,25);
+                Optional<List<TwitterPost>> resultListTwitterPost = Optional.ofNullable(twitter.getUserTimeline(page).stream()
+                        .map(s -> new TwitterPost(s.getText(),
+                                s.getUser().getName(),
```

Comment:
the s's should line up here, from . `s.getText()`

Reply:
also you could just have a constructor for TwitterPost that accepts whatever the type of `s` is then your map would be cleaner